tools/package_json_release_filter.jq

```diff
@@ -0,0 +1,32 @@
+# Copyright Google Inc. All Rights Reserved.
+#
+# Use of this source code is governed by an MIT-style license that can be
+# found in the LICENSE file at https://angular.io/license
+#
+# This filter combines a subproject package.json with the root package.json
+# and performs substitutions to prepare it for release. It should be called
+# with the --slurp argument and be passed the root pacakge.json followed by
+# the subproject package.json.
+#
+# See jq docs for filter syntax: https://stedolan.github.io/jq/manual/.
+
+.[0] as $root
+| .[1] as $proj
+
+# Get the fields from root package.json that should override the project
+# package.json, i.e., every field except the following
+| ($root
+    | del(.bin, .description, .dependencies, .name, .main, .peerDependencies, .optionalDependencies, .typings, .version, .private, .workspaces, .resolutions, .scripts, .["ng-update"])
+) as $root_overrides
+
+# Use the project package.json as a base and override other fields from root
+| $proj + $root_overrides
+
+# Combine keywords from both
+| .keywords = ($root.keywords + $proj.keywords | unique)
+
+# Remove devDependencies
+| del(.devDependencies)
+
+# Add engines
++ {"engines": {"node": "^12.20.0 || ^14.15.0 || >=16.10.0", "npm": "^6.11.0 || ^7.5.6 || >=8.0.0", "yarn": ">= 1.13.0"}}
```

Comment:
Should the engines really be configured here? personally that seems like a bad spot for me to have it placed. Would it make sense to have it somewhere else in a common spot for configuration of release output?

e.g. in Angular components we have a general config starlark file at the workspace top-level.

This could be a follow-up discussion, but that's just something I feel like could be improved IMO

Reply:
It looks like `engines` is set in the root `package.json`, but it was overridden in here: https://github.com/angular/angular-cli/blob/master/lib/packages.ts#L87

What if we just keep it in the root package.json and have that field override the project package.json in the filter?

Or would you prefer stamped substitutions like in the components repo?

Reply:
Good point. I think we cannot use the top-level `engines` field because that one prohobits the use of NPM, while actual consumers of the CLI should be fine to use NPM. Generally I like how we did the majority of versions within Angular Components (it's obviously not done for the engines, but I like that approach where a single place controls these common things; and it's not deeply hidden in some files).

Please feel free to leave this as is and discuss with @josephperrott and/or the CLI team.

Reply:
Yeah, I would love to do the substitutions if possible, but the follow up makes sense to me.

With the substitutions in place, we could have the template markers placed here in the JQ process and then use the starlark to consistently place the engine values.